tests/php/Core/Injector/InjectorTest.php

```diff
@@ -1062,6 +1062,11 @@ public function testExtendedExtensions()
                 SomeExtension::class,
             ]);
 
+        // Since we've changed config for the extension's class name after it has been loaded (via
```

Comment:
This change essentially invalidates this test. My read is that the feature ostensibly tested by this test isn't actually available. I'm surprised that it ever was but apparently it _was_ available in 4.1 (according to @robbieaverill if I understood a slack convo with him correctly).

I would recommend:
 * Removing this test altogether and noting in the docs that dynamically changing the extensions config option doesn't work, AND/OR
 * Putting this test into 4.1 and, if it passes, trying to work out what change broke it.

Essentially I'm recommending a _full_ revert of #8465.

Alternatively, we retain Extensible::flush_extra_methods_cache(), document that method as necessary to be called after dynamic modification of the extensions property, and call DataObject::flush_extra_methods_cache() here instead.

To be clear, I would retain the add-extensions / remove-extensions material that this PR re-introduces, which means that it's _not_ necessary to call DataObject::flush_extra_methods_cache in the setUp(). Rather, it only necessary to call if it you're making changes _between_ tests.

Reply:
I've had a look at this. I think the pass in 4.0, 4.1, 4.2 is bogus.

In particular, I think that it's only working if `Config::modify()` on Member extensions is called before any method-exist checks are called on Member.

Notably, this test fails (in 4.1 at least)
```diff
    /**
     * Tests that overloaded extensions work, see {@link Extensible::getExtensionInstance()}
     */
    public function testExtendedExtensions()
    {
+        $member = new Member();
+        $this->assertFalse($member->hasExtension(SomeExtension::class));
+
        Config::modify()
            ->set(Injector::class, SomeExtension::class, [
                'class' => SomeCustomisedExtension::class,
            ])
            ->merge(Member::class, 'extensions', [
                SomeExtension::class,
            ]);

        /** @var Member|SomeExtension $member */
        $member = new Member();
        $this->assertTrue($member->hasExtension(SomeExtension::class));
        $this->assertSame('bar', $member->someMethod());
    }
```

I haven't validated which change to SapphireTest causes this to fail – it's probably something about initialising Members – but my view is that this feature has therefore always been unreliable in 4.x and it's fine to simply document that it doesn't work.

Put another way, I think this might break tests going from 4.2 to 4.3, but those tests that were passing in 4.2 were relying on an unreliable code-path.

Reply:
@sminnee OK, thanks for following up - sounds like we're OK about reverting all of this and making it a documentation issue?